geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/ArrowAttributeReader.scala

```diff
@@ -102,26 +114,26 @@ object ArrowAttributeReader {
     * @param classBinding the explicit class binding of the attribute
     * @param vector the simple feature vector to read from
     * @param dictionary the dictionary for the attribute, if any
-    * @param precision geometry precision (ignored if the attribute is not a geometry)
+    * @param encoding encoding options
     * @return reader
     */
   def apply(bindings: Seq[ObjectType],
             classBinding: Class[_],
             vector: FieldVector,
             dictionary: Option[ArrowDictionary],
-            precision: GeometryPrecision): ArrowAttributeReader = {
+            encoding: SimpleFeatureEncoding): ArrowAttributeReader = {
     val accessor = vector.getAccessor
     dictionary match {
       case None =>
         bindings.head match {
+          case ObjectType.GEOMETRY => ArrowGeometryReader(vector, classBinding, encoding.geometry)
+          case ObjectType.DATE     => ArrowDateReader(vector, encoding.date)
```

Comment:
the primitive ones can just read the value directly from the underlying arrow vector - date and geometry are encoded differently and so need special handling. geometries are a bunch of nested vectors that need to be read using the geometry vector classes in geomesa-arrow-jts, dates are stored as longs or ints (for second precision) and need to be converted back to dates on read